### ISSUE TITLE
Condition false resources still evaluated

### ISSUE THREAD

**Issue by mjudeikis** [Azure/azure-resource-manager-schemas#1067]
Scenario:
We have complex ARM template where we have all resources set with `Conditions`. 
We deploy same template across multiple azure regions.
Some regions do not have certain resources enabled.
ARM template fails even when condition says false.

We suspect that resources are still evaluated and check before the condition is checked.

Example:
```
  {
            "name": "[concat(substring(parameters('acrResourceId'), add(lastIndexOf(parameters('acrResourceId'), '/'), 1)), '/', parameters('location'))]",
            "type": "Microsoft.ContainerRegistry/registries/replications",
            "location": "[parameters('location')]",
            "condition": "[and(parameters('fullDeploy'),parameters('acrReplica'))]",
            "apiVersion": "2019-05-01"
        },
```

In our case when we deploy to `eastus2euap` where ACR is not available we set `acrReplica` = false.
The template still fails with error what ACR replications is not available in the chosen region.

Expected results:
We would expect to be able to control deployment using conditions, and this should not error out.

**Comment by BernieWhite** [comment]
@mjudeikis True. The reason is that some constraints such as location and name are evaluated ahead of the condition property by ARM.

**Comment by ahahn95** [comment]
@BernieWhite Interesting, do you have a link to any docs that describe the order that properties are evaluated in a scenario like this?

**Comment by anthony-c-martin** [comment]
@bmoore-msft do you happen to know the status of this?

**Comment by BernieWhite** [comment]
> @BernieWhite Interesting, do you have a link to any docs that describe the order that properties are evaluated in a scenario like this?

@ahahn95 the order of evaluation is not documented publicly to my knowledge. Something to improve.

I know from experience that both name and location are evaluated early.

**Comment by ahahn95** [comment]
Thanks for the response.

I actually found that the issue that hung me up today was that a reference function was evaluating before the `condition` property does. I saw there is some documentation [here](https://docs.microsoft.com/en-us/azure/azure-resource-manager/templates/conditional-resource-deployment), but some more useful error messages would be nice.
